relays/substrate/src/millau_headers_to_rialto.rs

```diff
@@ -96,8 +101,71 @@ impl TargetClient<MillauHeadersToRialto> for RialtoTargetClient {
 	}
 }
 
+/// Maintainment procedure of Millau -> Rialto headers.
+struct MillauHeadersToRialtoMaintain {
+	millau_justifications: Arc<Mutex<JustificationsSubscription>>,
```

Comment:
Why does it have to be `Arc<Mutex` it doesn't seem to be sharing the state.

Reply:
`JustificationsSubscription::next()` requires `&mut self`.